experimental/add_to_app/flutter_module_using_plugin/lib/main.dart

```diff
@@ -0,0 +1,134 @@
+// Copyright 2019 The Flutter team. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:path_provider/path_provider.dart' as paths;
+
+void main() => runApp(MyApp());
+
+class MyApp extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      title: 'Flutter Module Title',
+      routes: {
+        '/': (context) => FullScreenView(),
+        '/mini': (context) => MiniView(),
+      },
+    );
+  }
+}
+
+class FullScreenView extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: Text('A Full-screen Flutter View'),
+      ),
+      body: const MiniView(showExit: true),
+    );
+  }
+}
+
+class MiniView extends StatefulWidget {
+  final bool showExit;
+
+  const MiniView({this.showExit = false});
+
+  @override
+  _MiniViewState createState() => _MiniViewState();
+}
+
+class _MiniViewState extends State<MiniView> {
+  int count = 0;
+
+  // Start off with a completed future so the FutureBuilder code below doesn't
+  // need to worry about null.
+  Future<void> saveFuture = Future.sync((){});
```

Comment:
Perhaps just Future.value(null)? Seems like there'd be less nested parenths/brackets that way :)

Reply:
`Future<void>` still mystifies me in some ways. I'm not sure why I didn't think to do that.